server/src/main/java/org/elasticsearch/script/field/ByteDocValuesField.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.script.field;
+
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.apache.lucene.util.ArrayUtil;
+import org.elasticsearch.index.fielddata.ScriptDocValues;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class ByteDocValuesField implements DocValuesField<Byte>, ScriptDocValues.Supplier<Long> {
+
+    protected final SortedNumericDocValues input;
+    protected final String name;
+
+    protected byte[] values = new byte[0];
+    protected int count;
+
+    private ScriptDocValues.Longs longs = null;
+
+    public ByteDocValuesField(SortedNumericDocValues input, String name) {
+        this.input = input;
+        this.name = name;
+    }
+
+    /**
+     * Set the current document ID.
+     *
+     * @param docId
+     */
+    @Override
+    public void setNextDocId(int docId) throws IOException {
+        if (input.advanceExact(docId)) {
+            resize(input.docValueCount());
+            for (int i = 0; i < count; i++) {
+                values[i] = (byte) input.nextValue();
+            }
+        } else {
+            resize(0);
+        }
+    }
+
+    protected void resize(int newSize) {
+        count = newSize;
+
+        assert count >= 0 : "size must be positive (got " + count + "): likely integer overflow?";
+        values = ArrayUtil.grow(values, count);
+    }
+
+    /**
+     * Returns a {@code ScriptDocValues} of the appropriate type for this field.
+     * This is used to support backwards compatibility for accessing field values
+     * through the {@code doc} variable.
+     */
+    @Override
+    public ScriptDocValues<Long> getScriptDocValues() {
+        if (longs == null) {
+            longs = new ScriptDocValues.Longs(this);
+        }
+
+        return longs;
+    }
+
+    @Override
+    public Long getInternal(int index) {
+        return (long) values[index];
+    }
+
+    /**
+     * Returns the name of this field.
+     */
+    @Override
+    public String getName() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Returns {@code true} if this field has no values, otherwise {@code false}.
+     */
+    @Override
+    public boolean isEmpty() {
+        return count == 0;
+    }
+
+    /**
+     * Returns the number of values this field has.
+     */
+    @Override
+    public int size() {
+        return count;
+    }
+
+    /**
+     * Returns an iterator over elements of type {@code T}.
+     *
+     * @return an Iterator.
+     */
```

Comment:
This comment doesn't seem necessary/helpful.

Reply:
Removed.